### PR TITLE
Sample config and README update.

### LINKED ISSUE #11: Basic documentation
For some more project information we need the following: 

- [x] README with setup and build instructions
- [x] Example config files for FRR, WiMoVE etc.

### PR DESCRIPTION
**Pull request by Sohn123**
Fixes #11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+
+##  State of this software
+
+:warning: This software is still work in progress and subject to change. :warning:
+
+Currently we support OpenWRT with the following architectures: `ramips-mt7621`, `mvebu-cortexa9`. We test our software on the following Access Points: `ZyXEL NWA50AX` and `Linksys WRT1900ACS`. If you need support for another OpenWRT architecture feel free to open an issue. We'll consider adding another build pipeline. If you encounter an issue or have a feature request please open a ticket here on github. 
+
+## Acknowledgements
+
+The article series of Vincent Bernat is a great starting point to learn about BGP EVPN with VXLAN. It helped us getting started so please check it out [Link](https://vincent.bernat.ch/en/blog/2017-vxlan-bgp-evpn)
```
Comment (dasGoogle) [inline README.md:31]:
```suggestion
The article series of Vincent Bernat is a great starting point to learn about BGP EVPN with VXLAN. It helped us get started so please check it out [here](https://vincent.bernat.ch/en/blog/2017-vxlan-bgp-evpn).
```

File: README.md [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
 
-## Formatting
+WiMoVE is built with standard network protocols, on top of open&#8209;source technology:
 
-Format the source files by running `make format`.
-Lint the source files by running `make lint`.
-Run both checks by doing `make precommit`.
+- The overlay networks use [BGP EVPN](https://www.rfc-editor.org/rfc/rfc8365.html) with [VXLAN](https://www.rfc-editor.org/rfc/rfc7348) encapsulation.
+- All BGP speakers run [FRRouting](https://frrouting.org/) to exchange EVPN routes.
+- The access points run [OpenWRT](https://openwrt.org/) with a custom, open&#8209;source daemon called [wimoved](https://github.com/wimove-oss/wimoved).
```
Comment (dasGoogle) [inline README.md:13]:
```suggestion
- The access points run [OpenWRT](https://openwrt.org/) with our custom, open&#8209;source daemon called [wimoved](https://github.com/wimove-oss/wimoved).
```

File: README.md [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
 
-As linting takes a long time, we recommend integrating `clang-tidy` into your editor.
+This solution allows for using commodity access points running OpenWRT for large&#8209;scale Wi&#8209;Fi deployments, even from different vendors.
+
+This repository contains the open&#8209;source daemon wimoved of the WiMoVE system. If you want to learn more about the architecture of the system, take a look at our [docs](https://wimove-oss.github.io/docs/)
```
Comment (dasGoogle) [inline README.md:17]:
```suggestion
This repository contains the open&#8209;source daemon wimoved of the WiMoVE system. If you want to learn more about the architecture of the system, take a look at our [documentation](https://wimove-oss.github.io/docs/architecture/).
```

File: README.md [hunk 4]
```diff
@@ -1,12 +1,31 @@
...
+
+## Setup
+
+Our documentation for the setup is still pretty early stage with some rough edges. If something doesn't work the way you expect or you get stuck please let us know, so we can improve our guides and help you with your issue. A hosted version of the setup guide is available [here](https://wimove-oss.github.io/docs/setup)
```
Comment (dasGoogle) [inline README.md:21]:
```suggestion
Our documentation for the setup is still pretty early stage with some rough edges. If something doesn't work the way you expect or you get stuck please let us know, so we can improve our guides and help you with your issue. A hosted version of the setup guide is available [here](https://wimove-oss.github.io/docs/setup).
```

**Review by dasGoogle — CHANGES_REQUESTED** [review]
Some minor things and some more punctuation

**Review by rgwohlbold — COMMENTED** [review]
I like the general introduction and the badges.
I think that overall, we need to get a clearer picture of what we want to have in this README vs in the documentation.

I think the line "This repository contains the open‑source daemon wimoved [...]" is somewhat hidden, I didn't really notice it when skimming the README. I suggest creating a "This repository" header that contains all information on this repository or links to the correct places.

The "Setup" part is redundant since it links to the same documentation we linked to above, same goes for the "State of this software". "Acknowledgements" are probably also a better fit for the `docs` repository.  

We should rather add some technical notes on how to work with the repository, like running, formatting, linting and testing the software. Maybe we should also consider having the development setup guide in this README, but deployment notes and general project documentation in the `docs` repo

**Comment by dasGoogle** [comment]
> I like the general introduction and the badges. I think that overall, we need to get a clearer picture of what we want to have in this README vs in the documentation.
> 
> I think the line "This repository contains the open‑source daemon wimoved [...]" is somewhat hidden, I didn't really notice it when skimming the README. I suggest creating a "This repository" header that contains all information on this repository or links to the correct places.
> 
> The "Setup" part is redundant since it links to the same documentation we linked to above, same goes for the "State of this software". "Acknowledgements" are probably also a better fit for the `docs` repository.
> 
> We should rather add some technical notes on how to work with the repository, like running, formatting, linting and testing the software. Maybe we should also consider having the development setup guide in this README, but deployment notes and general project documentation in the `docs` repo

I'd argue that many people will see this Readme/Repo as the entrypoint for the project, so we should also provide a basic overview of what the project does here and not just in the docs.

File: README.md [hunk 5]
```diff
@@ -1,12 +1,89 @@
...
+- Install [`prometheus-cpp`](https://github.com/jupp0r/prometheus-cpp).
+    - On Ubuntu, install the package `prometheus-cpp-dev`.
+    - On Arch Linux, the package is available in the AUR as `prometheus-cpp-git`.
+- Clone the repository by running `git clone https://github.com/WiMoVE-OSS/wimoved`.
+- Build the project by running `cmake .` followed by `make -j$(nproc)`.
+- Create a configuration file `wimoved.conf`
+
+```text
+hapd_sock=/var/run/hostapd/<interface>
+```
```
Comment (dasGoogle) [inline README.md:81]:
```suggestion
```text
hapd_sockdir=/var/run/hostapd/
```
```

File: README.md [hunk 6]
```diff
@@ -1,12 +1,89 @@
...
-Lint the source files by running `make lint`.
-Run both checks by doing `make precommit`.
+- The overlay networks use [BGP EVPN](https://www.rfc-editor.org/rfc/rfc8365.html) with [VXLAN](https://www.rfc-editor.org/rfc/rfc7348) encapsulation.
+- All BGP speakers run [FRRouting](https://frrouting.org/) to exchange EVPN routes.
+- The access points run [OpenWRT](https://openwrt.org/) with our custom, open&#8209;source daemon called [wimoved](https://github.com/wimove-oss/wimoved).
 
-As linting takes a long time, we recommend integrating `clang-tidy` into your editor.
+This solution allows for using commodity access points running OpenWRT for large&#8209;scale Wi&#8209;Fi deployments, even from different vendors.
+
+This repository contains the open&#8209;source daemon wimoved of the WiMoVE system. The daemon is responsible to handle events of hostapd and creating the VXLAN interfaces. The WiMoVE system consists of multiple other systems apart from the daemon. If you want to learn more about the architecture of the system or want to deploy your own setup, take a look at our [documentation](https://wimove-oss.github.io/docs/).
```
Comment (rgwohlbold) [inline README.md:17]: Language nitpick: _The daemon is reposible for handling hostapd events and creating VXLAN interfaces_

Comment (rgwohlbold) [reply README.md:17]: Instead of _the WiMoVE system_, I suggest putting _WiMoVE_

Comment (rgwohlbold) [reply README.md:17]: We should make clear that `wimoved` is supposed to run on the APs

Comment (rgwohlbold) [reply README.md:17]: Another language nitpick: _multiple other parts_ sounds better than _multiple other systems_ and is probably a little more accurate

File: README.md [hunk 7]
```diff
@@ -1,12 +1,89 @@
...
+
+##  Supported architectures
+
+Currently we support OpenWRT with the following architectures: `ramips-mt7621`, `mvebu-cortexa9`. We test our software on the following Access Points: `ZyXEL NWA50AX` and `Linksys WRT1900ACS`. If you need support for another OpenWRT architecture feel free to open an issue.
+
+## Development setup
+
+The development setup is an easy way to test out the WiMoVE daemon. Please be aware that after following this guide you do **not** have a working instance of the whole WiMoVE system. You can connect to the Wi-Fi but won't have any connectivity to other devices or the internet. Currently we only support Linux as development platform.
```
[the remark this answers is not in the document]
Comment (rgwohlbold) [reply README.md:25]: _won't have any connectivity_ -> _won't be connected_

File: README.md [hunk 8]
```diff
@@ -1,12 +1,89 @@
...
+- Install [`libnl`](https://github.com/thom311/libnl). On a recent Linux system, the corresponding package is probably already installed.
+- Install [`prometheus-cpp`](https://github.com/jupp0r/prometheus-cpp).
+    - On Ubuntu, install the package `prometheus-cpp-dev`.
+    - On Arch Linux, the package is available in the AUR as `prometheus-cpp-git`.
+- Clone the repository by running `git clone https://github.com/WiMoVE-OSS/wimoved`.
+- Build the project by running `cmake .` followed by `make -j$(nproc)`.
+- Create a configuration file `wimoved.conf`
+
+```text
+hapd_sock=/var/run/hostapd/<interface>
```
Comment (rgwohlbold) [inline README.md:80]: This is outdated, isn't it?

**Review by rgwohlbold — CHANGES_REQUESTED** [review]
I like this a lot better! I have some small language comments. We should also make sure that the configuration file is up-to-date

**Comment by Sohn123** [comment]
I really like those changes and think the readme is now ready. Let's wait for the other team members to proof read it. @dasGoogle @linaScience

File: README.md [hunk 9]
```diff
@@ -1,16 +1,100 @@
...
 Run both checks and the tests by doing `make precommit`.
+The coding guidelines are enforced via the CI pipeline.
 
-As linting takes a long time, we recommend integrating `clang-tidy` into your editor.
-
-## Testing
-
-Build the tests by running `make test`.
\ No newline at end of file
+As linting takes a long time, we recommend integrating clang-tidy into your editor.
```
Comment (linaScience) [inline README.md:100]: I would prefer 
```suggestion
As linting takes a long time, we recommend integrating `clang-tidy` into your editor.
```

File: README.md [hunk 10]
```diff
@@ -1,16 +1,100 @@
...
+Currently, we support OpenWRT with the architectures `ramips-mt7621` and `mvebu-cortexa9`.
+We test our software on the Access Point models `ZyXEL NWA50AX` and `Linksys WRT1900ACS`.
+If you need support for other OpenWRT architectures, feel free to open an issue.
+
+## Development setup
+
+The development setup is an easy way to test wimoved.
+Please be aware that after following this guide, you do **not** have a full WiMoVE installation.
+You can connect to the Wi-Fi but won't be connected to other devices or the Internet.
+Currently, we only support Linux as a development platform.
```
[the remark this answers is not in the document]
Comment (Sohn123) [reply README.md:38]: Sure I will add that link.

**Review by dasGoogle — APPROVED** [review]

**Pull request merged**